src/Provider/Keycloak.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/*!
+* Hybridauth
+* https://hybridauth.github.io | https://github.com/hybridauth/hybridauth
+*  (c) 2021 Hybridauth authors | https://hybridauth.github.io/license.html
+*/
+
+namespace Hybridauth\Provider;
+
+use Hybridauth\Adapter\OAuth2;
+use Hybridauth\Exception\InvalidApplicationCredentialsException;
+use Hybridauth\Exception\UnexpectedApiResponseException;
+use Hybridauth\Data;
+use Hybridauth\User;
+
+/**
+ * Keycloak OpenId Connect provider adapter.
+ *
+ * Example:
+ *         'Keycloak' => [
+ *             'enabled' => true,
+ *             'url' => 'https://your-keycloak', // depending on your setup you might need to add '/auth'
+ *             'realm' => 'your-realm',
+ *             'keys' => [
+ *                 'id' => 'client-id',
+ *                 'secret' => 'client-secret'
+ *             ]
+ *         ]
+ *
+ */
+class Keycloak extends OAuth2
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public $scope = 'openid profile email';
+
+    protected $url;
+
+    protected $realm;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $apiDocumentation = 'https://www.keycloak.org/docs/latest/securing_apps/#_oidc';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        parent::configure();
+
+        if (!$this->config->exists('url')) {
+            throw new InvalidApplicationCredentialsException(
+                'You must define a provider url'
+            );
+        }
+        $this->url = $this->config->get('url');
```

Comment:
Thank you for PR. It looks good for me. 

I was thinking about the following: 
do we really needed `$this->url` and `$this->realm` separately in the class properties, since they are available in the config and not used anywhere except `configure` method (at least for now)

I'd purpose to get rid of them.